library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.java

```diff
@@ -489,9 +491,20 @@ private void onSendMessage(String input, boolean isEdit) {
         binding.ivSend.setEnabled(false);
 
         if (isEdit) {
-            getEditMessage().setText(input);
-            getEditMessage().setAttachments(messageInputClient.getSelectedAttachments());
-            viewModel.getChannel().updateMessage(getEditMessage(),  new MessageCallback() {
+            Message message = getEditMessage();
+            message.setText(input);
+            if (messageInputClient.getSelectedAttachments() != null
+                    && !messageInputClient.getSelectedAttachments().isEmpty()){
+                List<Attachment>attachments = message.getAttachments();
+                for (Attachment attachment : messageInputClient.getSelectedAttachments()){
```

Comment:
By extracting `messageInputClient.getSelectedAttachments()` once  readability could be improved.